contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+- tolerations that tolerates taints with key `other_key` that has a `NoExecute` effect.
+
+##### Fine-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: ["mykey"]
+      operators: ["Equal"]
+      values: ["value"]
+      effects: ["NoSchedule"]
+    - keys: ["other_key"]
+      operators: ["Exists"]
+      effects: ["NoExecute"]
+```
+
+Here we allow tolerations in the following forms:
+- tolerations that tolerates all `PreferNoSchedule` taints with any value.
+- tolerations that tolerates taints based on any key existence with effect `NoSchedule`.
+Also note that this SchedulingPolicy does not allow tolerating NoExecute taints.
+
+##### Coarse-grain allowed tolerations
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: ["PreferNoSchedule"]
+    - keys: []
+      operators: ["Exists"]
+      effects: ["NoSchedule"]
+```
+an empty list of toleration allows all types of tolerations:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations: []
+```
+
+Which is equivalent to:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    tolerations:
+    - keys: []
+      operators: []
+      values: []
+      effects: []
+```
+
+#### default
+
+if no toleration is not specified, the following `SchedulingPolicy` will add a:
+- toleration for <mykey,value,NoSchedule> and <mykey,other_value,NoSchedule> Taints.
+- toleration for <other_key,NoExecute> Taint.
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  default:
+    tolerations:
+    - key: "mykey"
+      operator: "Equal"
+      values: ["value","other_value"]
+      effect: "NoSchedule"
+    - key: "other_key"
+      operator: "Exists"
+      effect: "NoExecute"
+```
+note: an empty array of toleration is not a valid syntax for default toleration.
+
+### Priority classes
+
+Priority class usage can be regulated using fine-grain rules with `priorityClasseName` field.
+
+##### Only allow a single priority class
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  required:
+    priorityClasseNames: ["high-priority"]
+  default:
+    priorityClasseName: "high-priority"
+```
+In this example, only the `high-priority` PriorityClass is enforced by default.
+
+##### Allow all priorities
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    priorityClasseNames: []
+```
+In this example, all priority classes are allowed, but not mandatory.
+
+Note: an empty list of required priorityClasseNames is considered as invalid
+```yaml
+Kind: SchedulingPolicy
+spec:
+  required:
+    priorityClasseNames: []
+```
+
+
+### Node Selector
+
+The `nodeSelector` fields in `required`, `default` and `allowed` sections allow to precise what nodeSelectors are mandatory, possible and may provide default values if not set. As for other components, `required` nodeSelectors are automatically considered as allowed.
```

Comment:
I recommend not treating nodeSelector as distinct from node affinity here.

Reply:
@bgrant0607 - what do you mean by same ? same behaviour (i.e empty vs unset) ? or treating them on a same sub-type ?